pull-requests.md

```diff
@@ -1,63 +1,148 @@
-# The art of the pull request
+# Why do we review PRs?
+Code review is an essential part of the continuous deployment pipeline. It enables a second pair of eyes to:
+- Examine your change and find any problems you may have missed
+- Identify opportunities to optimise or improve your change
+- Provide a separate perspective from someone who may also have to work on your code
+- Highlight which parts might be difficult to understand for future developers
+- Coordinate across subject matter experts and teams, in order to maintain awareness and quality control over complex projects
 
-It is worth reading [Github's pull request conventions](https://github.com/blog/1943-how-to-write-the-perfect-pull-request) as this seem to cover most of the basics.
+This last aspect is especially important in mono-repos as:
+- Multiple teams might be submitting changes to the repository
+- Some code changes may have non-obvious impact. For example, whether a code change directly or indirectly impacts data security. Understanding the impact of a change can be hard without the insight and input of a subject matter expert.
 
-Pull requests are:
+# The art of raising a PR
+It is worth reading [GitHub's blog post "How to write the perfect pull request"](https://github.com/blog/1943-how-to-write-the-perfect-pull-request), and [How to Make Your Code Reviewer Fall in Love with You](https://mtlynch.io/code-review-love/) as these are offer excellent advice on Pull Requests (PRs).
 
-## Small and frequent
+The following are generally applicable to PRs, but are not hard rules.
 
-A good pull request is easy to understand. There should be a manageable number of changes in each pull request. Try to make your pull request as small as possible while still advancing the functionality of the product.
+## Small
+A PR should be easy to understand, and address a single concern.
 
-Aim to deploy a pull request before submitting the next one. Frequently small deployments of functionality are easier to review and easier to understand in production.
+There should be a manageable number of changes in a PR. Try to make your PR as small as possible while still advancing the functionality of the product.
 
-* Don't mix multiple changes into the same pull request.
-* If you are doing bulk moves and deletions only have one kind of operation in a pull request
+If a PR is too complex, it is much easier for issues to be missed. In addition if trying to understand an aspect of the code change in the future, it becomes harder to understand if buried in a very large and complex PR.
+
+Aim to deploy a PR before submitting the next one. Frequently small deployments of functionality are easier to review and easier to understand in production.
```

Comment:
I refer to these as Stacked PRs. There are a few good reads about this (for example [this](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/) or [this](https://blog.logrocket.com/using-stacked-pull-requests-in-github/)).

Although I often use them, I do sometimes find them tricky. For example, if the history of a parent PR is rewritten, the child PR needs to be rebased so that it merges cleanly, however the rebase isn't always trivial.

For this reason, I'd be tempted to avoid _recommending_ them here, but I do agree, it'll be nice to make a note of this practice somewhere.

Reply:
agree they're useful judiciously, but i find they often create noise for reviewers, even though they make a lot of sense the to PR raiser (which one do i need to review? which of these other PRs that are open and look like the need reviewing don't need reviewing yet?)

in a small team where everyone knows what's going on that's not a huge overhead. but in larger codebases, esp with lots of external contributors, it can be cumbersome